fhir-path/src/main/java/com/ibm/fhir/path/evaluator/FHIRPathEvaluator.java

```diff
@@ -1457,5 +1462,59 @@ public void unsetExternalConstant(String name) {
         public boolean hasExternalConstant(String name) {
             return externalConstantMap.containsKey(name);
         }
+
+        /**
+         * Set the constraint currently under evaluation
+         *
+         * @param constraint
+         *     the constraint currently under evaluation
+         */
+        public void setConstraint(Constraint constraint) {
+            this.constraint = constraint;
+        }
+
+        /**
+         * Unset the constraint currently under evaluation
+         */
+        public void unsetConstraint() {
+            constraint = null;
+        }
+
+        /**
+         * Get the constraint currently under evaluation
+         *
+         * @return
+         *     the constraint currently under evaluation if exists, otherwise null
+         */
+        public Constraint getConstraint() {
+            return constraint;
+        }
+
+        /**
+         * Indicates whether this evaluation context has an associated constraint
+         *
+         * @return
+         *     true if this evaluation context has an associated constraint, otherwise false
+         */
+        public boolean hasConstraint() {
+            return constraint != null;
+        }
+
+        /**
+         * Get the list of issues that occurred during evaluation
+         *
+         * @return
+         *     the list of issues that occurred during evaluation
+         */
+        public List<Issue> getIssues() {
+            return issues;
+        }
```

Comment:
is it only warnings, or might we expect errors here too?

Reply:
I made it flexible enough to support both.

Reply:
but currently, an error will result in a thrown exception, right?  and presumably an empty list of issues?

Reply:
Added javadoc to clarify that these are "supplemental issues" used to convey additional information to the client.